core/src/saros/net/stun/internal/StunServiceImpl.java

```diff
@@ -140,6 +141,9 @@ public void run() {
                 new InetSocketAddress(stunAddress, stunPort),
                 new InetSocketAddress(localAddress, 0),
                 timeout);
+      } catch (PortUnreachableException e) {
+        log.info("Aborted stun discovery - ICMP port unreachable: " + e.getMessage());
```

Comment:
This exception is raised when you receive an ICMP packet stating the the port (in this case UDP) is not reachable.

The message reads like the ICMP port is not reachable. Guess you forgot a delimiter after ICMP.

For clarification the code here is executed on every interface, so I would suggest.

1. to log this a warning or debug.
2. the message should contain the interface that was used.

Reply:
I set us as 'info' as I am not sure how important this information is. If its just our default stun discovery that always fails, this can be debug. But, if the user sets up a stun server and the connection is still refused, they should at least be able to find something about it in the log files (as far as I know, debug logging is disabled on the shipped Saros versions).

> Guess you forgot a delimiter after ICMP.

What exactly do you mean with this?
I know that the error message is somewhat redundant as it also reads "ICMP Port Unreachable", but I still decided to keep it in, in case there are other cases where the message might be useful. I probably should just drop the hard-coded message part of the log instead.